.github/workflows/ci.yaml

```diff
@@ -0,0 +1,46 @@
+---
+name: DNF CI
+on: pull_request_target
+
+jobs:
+  integration-tests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    container:
+      image: fedora:latest
+      options: --privileged
+    steps:
+      - name: Install API token for Copr
+        id: copr
+        run: |
+          # hack: Github replaces secrets with *** in the whole output (even in
+          # e.g. Copr URLs printed by rpm-gitoverlay). If there's a comment (#)
+          # at the end of the secret (e.g.  "rpmsofwaremanagement #"), this
+          # will clean it up and since it is no longer the whole secret being
+          # printed, Github won't hide it anymore.
+          COPR_USER=${{secrets.COPR_USER}}
+          if [ -z "$COPR_USER" ]; then echo "COPR_USER secret is required to run the CI."; exit 1; fi
+          echo "::set-output name=copr-user::$COPR_USER"
+
+          mkdir -p "$HOME/.config"
+          echo "${{secrets.COPR_API_TOKEN}}" > "$HOME/.config/copr"
+
+      - name: Check out ci-dnf-stack
+        uses: actions/checkout@v2
+        with:
+          repository: rpm-software-management/ci-dnf-stack
+
+      - name: Setup CI
+        uses: ./.github/actions/setup-ci
+
+      - name: Check out sources
+        uses: actions/checkout@v2
+        with:
+          path: gits/${{github.event.pull_request.head.repo.name}}
+          ref: ${{github.event.pull_request.head.sha}}  # check out the PR HEAD
+          fetch-depth: 0
+
+      - name: Run CI
+        uses: ./.github/actions/run-ci
```

Comment:
Apparently you forgot to git add this file?

The security of this hinges on what run-ci does with the untrusted checkout in gits/...  . If it does not use any code from that checkout, but calls everything from the known-good ci-dnf-stack repo, then using `pull_request_target` seems fine. Or alternatively, if it does untrusted operations inside a container (without passing in any secrets).

Reply:
> Apparently you forgot to git add this file?

The file is in the linked ci-dnf-stack PR: https://github.com/rpm-software-management/ci-dnf-stack/pull/952

The way the repos are checked out here is kind of "upside down", but I need to check out the repo with the shared CI code first and then build from there...

> The security of this hinges on what run-ci does with the untrusted checkout in gits/... . If it does not use any code from that checkout, but calls everything from the known-good ci-dnf-stack repo, then using `pull_request_target` seems fine. Or alternatively, if it does untrusted operations inside a container (without passing in any secrets).

Yes, that's what it does. The untrusted code in `gits/...` is basically packaged in Copr and installed into a container.

Reply:
Oh, of course, I didn't look at the first actions/checkout carefully enough. Sorry, we are probably repeating some parts of our email discussion here, but that's maybe ok for reviewers and future reference. This looks all fine to me then. Thanks!